lib/active_support/cache/redis_store.rb

```diff
@@ -55,6 +58,8 @@ def initialize(*addresses)
                   pool_options[:timeout] = options[:pool_timeout] if options[:pool_timeout]
                   @pooled = true
                   ::ConnectionPool.new(pool_options) { ::Redis::Store::Factory.create(*addresses) }
+                elsif @options[:redis]
```

Comment:
Love the idea, but can we use a different option name here? I'd prefer something like `:client`, so it's a little more explicit as to what the option is used for.